sphinx/writers/html5.py

```diff
@@ -345,13 +345,24 @@ def visit_comment(self, node: Element) -> None:
 
     # overwritten
     def visit_admonition(self, node: Element, name: str = '') -> None:
-        self.body.append(self.starttag(
-            node, 'div', CLASS=('admonition ' + name)))
+        if node.get('collapsible'):
+            if node.get('open'):
+                self.body.append(
+                    self.starttag(node, 'details', CLASS=('admonition ' + name), open='open')
+                )
+            else:
+                self.body.append(self.starttag(node, 'details', CLASS=('admonition ' + name)))
```

Comment:
The repetition here could be reduced with a dict expanded for kwargs to `starttag()`.